app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,66 @@
+class QuestionsController < ApplicationController
+
+  def index
+    @questions = Question.last(10)
+    # @hot_questions = Question.top_vote_day
+    # @week_questions = Question.top_vote_week
+    # @month_questions = Question.top_vote_month
+    #TODO build question selection methods in question model
+  end
+
+
+
+  def new
+    @question = Question.new
+  end
+
+  def show
+    if current_user
+      @question = Question.find(params[:id])
+      @answer = Answer.new
+      @answers = @question.answers
+      @score = @question.score
+      @question_comments = @question.comments
+      @comment = Comment.new
+      @vote = Vote.new
```

Comment:
Avoid setting multiple ivars for the view. Let the view use @question.answers rather than doing @answers = @question.answers.

In the controller you can do @question = Question.includes(:answers).find(params[:id]) to signal you'll be using the answers.

Setting ivars might suggest to a reader that they carry a set of answers other than this question's answers (otherwise why wouldn't you just have used @question.answers.)